### PR TITLE
Documentation skeleton

### LINKED ISSUE #1174: [RFC] Documentation structure
## Need

Backstage is currently poorly documented. The existing documentation lacks a clear structure making it hard to get an overview. This RFC outlines a suggested structure for the docs we want to have. It does not propose the actual contents.

## Proposed solution

Suggested TOC:

- What is Backstage?
  - Introduction
  - Main concepts
  - Architecture
- Getting started
  - Installation
  - Running Backstage locally
  - Production deployments
    - Create an App
    - App configuration
      - Configuring App with plugins
      - Customize the look-and-feel of your App
    - Deployment scenarios
- Features
  - Software Catalog
    - System model
    - YAML File Format
    - Extending the model
    - External integrations
  - Scaffolding service
    - Configure templates
    - Adding templates
  - Authentication and identity
- Plugins
  - Creating a new plugin
  - Developing a plugin
  - Backends and APIs
    - Proxying
    - Backstage backend plugin
    - Existing API
  - Designing for Backstage
    - Backstage Design Language System (DLS)
    - Storybook -- reusable UI components
    - Figma resources
  - Testing
  - Publishing
    - Open source and NPM
    - Private/internal (non-open source)
- API references
  - JavaScript/TypeScript API
  - Backend APIs
- Tutorials
  - How to do X
  - How to do Y
- Architectural Decision Records (ADRs)
- Contribute
- Support
- FAQ

## Implementation

These docs should be stored in the `/docs` folder and be published to `backstage.io/docs`.

## Questions

* Should "Service Catalog", "Identity" and "Scaffolding service be grouped together?
* Do we need a more clear separation between frontend and backend?

### PR DESCRIPTION
**Pull request by stefanalund**
## Hey, I just made a Pull Request!

Adds initial skeleton for new documentation structure. We plan to use the Backstage [TechDocs feature](https://github.com/spotify/backstage/tree/master/plugins/techdocs) to host our own documentation (meta, I know).

Example of how this will (almost) look:

Screenshot 2020-07-14 at 12 52 40

Fixes https://github.com/spotify/backstage/issues/1174

#### :heavy_check_mark: Checklist

- [x] Screenshots attached (for UI changes)
- [x] Relevant documentation updated
- [x] Prettier run on changed files

### CODE REVIEW

**Review by benjdlambert — APPROVED** [review]

File: docs/README.md [hunk 1]
```diff
@@ -1,10 +1,83 @@
-# Documentation
+# Documentation structure
```
Comment (Rugvip) [inline docs/README.md:2]: Maybe a small disclaimer here about thing being WIP and a lot is gonna be missing? :grin:

Comment (stefanalund) [reply docs/README.md:2]: Fixed @Rugvip

**Comment by chrsmith** [comment]
Hello! This PR added some placeholder documents, and I was wondering if there is a new location where I can browse some of the missing files.

For example, [/docs/getting-started/configure-app-with-plugins.md](https://github.com/spotify/backstage/blob/master/docs/getting-started/configure-app-with-plugins.md) was added in this PR and is 0-bytes in length. Was this an error, or just stubbing out the file while hooking things up to look awesome in the "TechDocs" plugin?

Ultimately, I'm just trying to get an instance of Backstage setup at my company, and better understand the platform and how it could help us manage cloud infrastructure. And to that end the docs haven't been all that helpful -- especially for the newer features like setting up/configuring "Service Catalogs" (which is something that we at [Pulumi](www.pulumi.com) are especially interested in).

Is there a better or feature branch to get the latest docs? Or should I hop on the Discord channel and just ask a bunch of basic troubleshooting type questions?

**Comment by dtuite** [comment]
@chrsmith Does this help? https://github.com/spotify/backstage/blob/master/docs/overview/architecture-overview.md

**Comment by chrsmith** [comment]
Thanks @dtuite , it does. One of the things that had me stumped was how to actually add/configure plugins, which evidently needs to be done by editing the App code:
https://github.com/spotify/backstage/blob/master/docs/overview/architecture-overview.md#installing-plugins

So I presume the next step would be to add some plugins, such as the Service catalog. And just to correct my own understanding, "Service Catalog" isn't a top-level component, in that there isn't any notion of "currently selected service" when viewing the UI of a plugin, correct? E.g. you cannot drill into the "frontend team's" services, and then show the Code Coverage plugin scoped to just that team?

Instead, each plugin operates independently at a high-level, with no state/context sharing between them. Do I have that right? However, presumably a plugin could use the Service Catalog API to query services, and provide a way to drill into a specific Service though...

Anyways, thank you for the help. Hopefully I'll have some meaningful feedback and/or contribution soon.

**Comment by dtuite** [comment]
@chrsmith 

> which evidently needs to be done by editing the App code

Either that, or fork the repo and just run it directly.

> in that there isn't any notion of "currently selected service" when viewing the UI of a plugin, correct? E.g. you cannot drill into the "frontend team's" services, and then show the Code Coverage plugin scoped to just that team?

Not yet, but the internal version of Backstage does have that and it will be added to the OSS version eventually.

**Pull request merged**